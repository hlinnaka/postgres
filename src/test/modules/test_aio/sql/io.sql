SELECT count(*) FROM tbl_a WHERE ctid = '(1, 1)';

SELECT corrupt_rel_block('tbl_a', 1);

-- FIXME: Should report the error
SELECT redact($$
  SELECT read_corrupt_rel_block('tbl_a', 1);
$$);

-- verify the error is reported
SELECT redact($$
  SELECT count(*) FROM tbl_a WHERE ctid = '(1, 1)';
$$);
SELECT redact($$
  SELECT count(*) FROM tbl_a;
$$);
