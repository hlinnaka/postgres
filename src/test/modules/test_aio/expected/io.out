SELECT count(*) FROM tbl_a WHERE ctid = '(1, 1)';
 count 
-------
     1
(1 row)

SELECT corrupt_rel_block('tbl_a', 1);
 corrupt_rel_block 
-------------------
 
(1 row)

-- FIXME: Should report the error
SELECT redact($$
  SELECT read_corrupt_rel_block('tbl_a', 1);
$$);
 redact 
--------
 t
(1 row)

-- verify the error is reported
SELECT redact($$
  SELECT count(*) FROM tbl_a WHERE ctid = '(1, 1)';
$$);
NOTICE:  wrapped error: invalid page in block 2 of relation base/<redacted>
 redact 
--------
 f
(1 row)

SELECT redact($$
  SELECT count(*) FROM tbl_a;
$$);
NOTICE:  wrapped error: invalid page in block 2 of relation base/<redacted>
 redact 
--------
 f
(1 row)

